File: kotlin/hamming/src/main/kotlin/Hamming.kt
class Hamming {

    companion object {
        fun compute(firstSequence: String, secondSequence: String): Int {
            require(firstSequence.length == secondSequence.length) { "left and right strands must be of equal length." }
            return firstSequence.indices.count { firstSequence[it] != secondSequence[it] }
        }
    }
}

File: go/hamming/hamming.go
package hamming

import "errors"

const testVersion = 5

// Distance calculates the Hemming distance between DNA strands a & b.
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("invalid strands. Must be of equal length")
	}

	var numDifferences int

	for i := range a {
		if a[i] != b[i] {
			numDifferences++
		}
	}

	return numDifferences, nil
}
